tests/PHPStan/Analyser/NodeScopeResolverTest.php

```diff
@@ -2478,11 +2478,11 @@ public function dataTypeElimination(): array
 				'$intOrTrue',
 				"'yetYetAnotherIntForSure';",
 			],
-			[
-				'TypeElimination\Foo|null',
-				'$fooOrStringOrNull',
-				"'fooOrNull';",
-			],
+//			[
+//				'TypeElimination\Foo|null',
+//				'$fooOrStringOrNull',
+//				"'fooOrNull';",
+//			],
```

Comment:
Because the refactoring break this behavior at this point. The behavior has been restored back in the next commit https://github.com/phpstan/phpstan/pull/278/commits/f05893f6fa66af73e075cc88660b5a3935ddb35f